src/ModuleApi.ts

```diff
@@ -46,14 +47,14 @@ export interface ModuleApi {
 
     /**
      * Opens a dialog in the client.
-     * @param title The title of the dialog
+     * @param titleOrOptions The title of the dialog, or an object that configures the dialog.
      * @param body The function which creates a body component for the dialog.
      * @param props Optional props to provide to the dialog.
      * @returns Whether the user submitted the dialog or closed it, and the model returned by the
      * dialog component if submitted.
      */
     openDialog<M extends object, P extends DialogProps = DialogProps, C extends DialogContent<P> = DialogContent<P>>(
-        title: string,
+        titleOrOptions: string | ModuleUiDialogOptions,
         body: (props: P, ref: React.RefObject<C>) => React.ReactNode,
```

Comment:
Just a comment: thb, I unsure why this was not defined as `BodyComponent: React.Component<P>`. Then the users wouldn't need to work with the `ref` and also the `props` in here would be more clear.

```tsx
// could do this:
openDialog("title", DialogBody, { greeting: 'Hi!' });

// instead of this:
openDialog(
  "title",
  (props, ref) => (
    <DialogBody
      ref={ref}
      {...props}
      // why not add greeting="Hi!" here instead of passing it in the next line?
    />
  ),
  { greeting: 'Hi!' },
);
```

But nothing I want to change (or even do without breaking compatibility).

Reply:
yeah, no idea :(